vignettes/extending-diseasystore-example.Rmd

```diff
@@ -0,0 +1,577 @@
+For the expiration date of the information, we could either use `NA`, since the birth date is always valid, or
+we could use the date of death, if the individual has died.
+In this implementation we use the latter, since this allows for an easier implementation of the age of the individuals
+later on.
+
+Notice that `valid_until` is set as the date after death. The validity period of the features in `diseasystore` is
+the interval $t \in \left[\textrm{valid_from}, \textrm{valid_until}\right)$.
+
+After computing the feature, we filter to ensure that the data is returned only for the requested time period.
+
+Finally, we also how the feature can be summarised. That is, we need to specify the type of `key_join` the feature
+should use.
+
+To illustrate how to determine the type of `key_join`, we imagine that we want to determine the number of people who
+are born on each date. Given our data set, we could perform the following operations to summarize the data:
+
+```{r key_join_birth}
+simulist_data |>
+  dplyr::count(lubridate::year(.data$birth))
+```
+
+That is, our feature has a record for each individual, and to summarise the feature, we need to `count` the individuals
+within each group. This is different from semi-aggregated data where we would `sum` the values within each group.
+
+
+## The `sex` `FeatureHandler`
+We continue by implementing the `sex` feature.
+
+Like the `birth` feature, this is a simple feature that we can mostly extract directly from the data set:
+
+```{r feature_handler_sex, eval = FALSE}
+private = list(
+  ...
+
+  # The "sex" feature simply stores the sex from the simulist data
+  simulist_sex = FeatureHandler$new(
+    compute = function(start_date, end_date, slice_ts, source_conn, ds, ...) {
+
+      out <- simulist_data |>
+        dplyr::right_join( # Join with birth data to validity period
+          ds$get_feature("birth", start_date, end_date, slice_ts),
+          by = c("id" = "key_pnr"),
+          copy = TRUE
+        ) |>
+        dplyr::transmute(
+          "key_pnr" = .data$id,
+          "sex" = dplyr::if_else(.data$sex == "m", "Male", "Female"),
+          .data$valid_from, .data$valid_until # Use values from birth feature
+        )
+
+      # No need to filter to ensure the data is only for the requested time period.
+      # Since we right join with the birth feature, the validity period is already filtered.
+
+      return(out)
+    },
+    key_join = key_join_count
+  ),
+
+  ...
+)
+```
+
+Notice that we again create a new instance of `FeatureHandler` and assign it to the `simulist_sex` field in the
+private list.
+
+The computation of the feature is mostly straight forward, but we would like to use the validity period computed
+in the `birth` feature (the same logic applies for the validity period of the "sex" information as the "birth"
+information)
+
+Therefore, we immediately see a difference in the function signature of the `compute()` function, where we no include
```

Comment:
Therefore, we immediately see a difference in the function signature of the `compute()` function, where we don't include